social_protection/models.py

```diff
@@ -116,6 +117,84 @@ def get_queryset(cls, queryset, user):
         return queryset.filter(individual__in=individual_queryset)
 
 
+class AbstractProjectTimeEntry(core_models.HistoryBusinessModel):
+    """
+    Base model for recording daily percent completion for beneficiaries in projects.
+    Subclasses must implement `_get_beneficiary_instance()`.
+    """
+    # Sequential workday number within the project (1..working_days).
+    day_number = models.PositiveSmallIntegerField(validators=[MinValueValidator(1)])
+
+    # Percentage of work completed for this day (0–100).
+    percent_complete = models.IntegerField(
+        validators=[MinValueValidator(0), MaxValueValidator(100)],
+    )
+
+    class Meta:
+        abstract = True
+
+    def _get_beneficiary_instance(self):
+        """
+        Subclasses must override this to return the correct beneficiary object
+        (either Beneficiary or GroupBeneficiary).
+        """
+        raise NotImplementedError("_get_beneficiary_instance() must be implemented in subclass")
+
+    def clean(self):
+        beneficiary = self._get_beneficiary_instance()
+        project = getattr(beneficiary, "project", None)
+
+        if not project:
+            raise ValidationError(
+                _("Beneficiary must be assigned to a project before recording time entries.")
+            )
+
+        if self.day_number > project.working_days:
```

Comment:
Can this condition be made consistent with the one at `services.py`: 

i.e. `if not 1 <= day <= project.working_days:`